bundles/org.eclipse.e4.ui.workbench/src/org/eclipse/e4/ui/workbench/IWorkbench.java

```diff
@@ -157,6 +157,12 @@ public interface IWorkbench {
 	/**
 	 * restart the workbench
 	 *
+	 * <p>
+	 * If the <code>--launcher.noRestart</code> argument is given, this method
+	 * should indicate to the user that the restart can't be completed, and it
+	 * should then return <code>false</code>.
+	 * </p>
+	 *
```

Comment:
This `IWorkbench` interface, similar to the other one is marked with `@noimplement`, which means that only Eclipse itself is supposed to implement it and downstream consumers should not implement it (it is not enforced, but strongly discouraged not to do it).

Therefore I suggest to not mention the launcher option here and at the other `IWorkbench` and just leave these two interfaces unchanged.
In general I assume that this option is only used rarely in UI applications (it's actually not intended to be used then) and therefore we should only add reference to it where really necessary, to no complicate the usual cases.